### PR TITLE
Cleanup ErcMsgList

### PR DESCRIPTION
**Pull request by rnestler**
See #93

While looking through the code base to get familiar with it I will fix some of those.

### CODE REVIEW

**Comment by ubruhin** [comment]
I always use the same order of these methods, so I would consider it as our guidelines:

```c++
public:
  // Constructors / Destructor
  [Default Constructor]
  [Copy Constructor]
  [Custom Constructors]
  [Destructor]

  // getters, setters, ...

  // Operator Overloadings
  [Assignment Operator]
  [Other Operators]
```
Example: [netclass.h](https://github.com/rnestler/LibrePCB/blob/cb13a80110c0516c140bad3ce909cfb65733a3b7/libs/librepcbproject/circuit/netclass.h)

**Comment by rnestler** [comment]
> I always use the same order of these methods, so I would consider it as our guidelines:

I don't like the guideline that much since I consider the copy constructor, assignment operator and destructar as a unit which belongs together. See [Rule of Three](https://en.wikipedia.org/wiki/Rule_of_three_(C%2B%2B_programming)).

But consistency is more important.

File: libs/librepcbproject/erc/ercmsglist.h [hunk 1]
```diff
@@ -54,9 +54,12 @@ class ErcMsgList final : public QObject, public IF_XmlSerializableObject
     public:
 
         // Constructors / Destructor
+        ErcMsgList() = delete;
+        ErcMsgList(const ErcMsgList& other) = delete;
         explicit ErcMsgList(Project& project, bool restore, bool readOnly, bool create) throw (Exception);
         ~ErcMsgList() noexcept;
 
+
```
Comment (rnestler) [inline libs/librepcbproject/erc/ercmsglist.h]: Should I remove this empty line?

Comment (ubruhin) [inline libs/librepcbproject/erc/ercmsglist.h]: It would be perfect if you moved this empty line from here to above the "signals:" statement :)

Comment (rnestler) [inline libs/librepcbproject/erc/ercmsglist.h]: Done

**Pull request merged**